cmd/av/stack_tree.go

```diff
@@ -68,14 +68,66 @@ func printStackTree(repo *git.Repo, branches map[string]meta.Branch, currentBran
 		fmt.Printf("%s<ERROR: unknown branch: %s>\n", indent, root)
 		return
 	}
+
+	branchInfo := getBranchInfo(repo, branch)	
+
 	if currentBranch == branch.Name {
 		_, _ = fmt.Print(
-			indent, colors.Success("* "), colors.Success(branch.Name), "\n",
+			indent, colors.Success("* "), colors.Success(branch.Name), " ", colors.Faint(branchInfo), "\n",
 		)
 	} else {
-		_, _ = fmt.Printf("%s%s\n", indent, branch.Name)
+		_, _ = fmt.Print(indent, branch.Name, " ", colors.Faint(branchInfo), "\n")
 	}
 	for _, next := range branch.Children {
 		printStackTree(repo, branches, currentBranch, next, depth+1)
 	}
 }
+
+func getBranchInfo(repo *git.Repo, branch meta.Branch) string {
+	var branchInfo string
+
+	parentStatus := getParentStatus(repo, branch)
+	upstreamStatus := getUpstreamStatus(repo, branch)
+
+	branchStatus := strings.Trim(fmt.Sprintf("%s %s", parentStatus, upstreamStatus), " ")
+	if branchStatus != "" {
+		branchInfo = fmt.Sprintf("(%s)", branchStatus)
+	}
+
+	if branch.PullRequest != nil && branch.PullRequest.Permalink != "" {
+		branchInfo = branch.PullRequest.Permalink + " " + branchInfo
+	}
+
+	return branchInfo
+}
+
+// Check if branch is up to date with the parent branch.
+// This is doing `git diff <parentBranch> <givenBranch>`
+func getParentStatus(repo *git.Repo, branch meta.Branch) string {
```

Comment:
I thought that "need sync" would mean it needs to be rebased (== need av stack sync). In order to tell that, you would need to check if mergebase(parent, branch) == parent.

Reply:
Also, if you do that, it'd be nice to compare with the remote tracking branch for trunks. i.e. in a pseudo code:

```
if branch.Parent.Trunk {
    return mergebaseCheck("origin/" + branch.Parent.Name, branch.Name)
}
return mergebaseCheck(branch.Parent.Name, branch.Name)
```

Reply:
ok updated to use mergebase. should i return empty string or "needs sync" in case of an error?

Reply:
Usually better to return `(string, error)`